pkg/controllers/addonmanagement/managedcluster.go

```diff
@@ -326,18 +377,31 @@ func (c *clusterController) applyWorkloadCluster(ctx context.Context, workspaceH
 			"apiVersion": "workload.kcp.dev/v1alpha1",
 			"kind":       "WorkloadCluster",
 			"metadata": map[string]interface{}{
-				"name": clusterName,
+				"name":   clusterName,
+				"labels": labels,
 			},
 			"spec": map[string]interface{}{
 				"kubeconfig": "",
 			},
 		},
 	}
-	if _, err := dynamicClient.Resource(clusterGVR).Create(ctx, workloadCluster, metav1.CreateOptions{}); err != nil {
-		return err
-	}
 
-	c.eventRecorder.Eventf("KCPWorkloadClusterCreated", "The KCP workload cluster %s is created in workspace %s", clusterName, workspaceConfig.Host)
+	if wc == nil {
+		if _, err := dynamicClient.Resource(clusterGVR).Create(ctx, workloadCluster, metav1.CreateOptions{}); err != nil {
+			return err
+		}
+
+		c.eventRecorder.Eventf("KCPWorkloadClusterCreated", "The KCP workload cluster %s is created in workspace %s", clusterName, workspaceConfig.Host)
+	} else {
+		// TODO - Does this need to be a PATCH? (Probably.) Are we going to be in conflict with updates being made on the object by KCP? Assume
+		// we control the spec and KCP controls the status.
+		workloadCluster.SetResourceVersion(wc.GetResourceVersion())
```

Comment:
for update labels, we may use `"github.com/openshift/library-go/pkg/operator/resource/resourcemerge"` to merge labels, maybe like

```
workloadClusterLabels := wc.GetLabels()
modified := resourcemerge.BoolPtr(false)
resourcemerge.MergeMap(modified, &workloadClusterLabels, workloadClusterLabels)

if *modified {
    ws.SetLabels(workloadClusterLabels)
    if _, err := dynamicClient.Resource(clusterGVR).Update(ctx, ws, metav1.UpdateOptions{}); err != nil {
	return err
    }
}

```

Reply:
We haven't talked about whether we'd expect an admin to be allowed to edit the workloadcluster directly in KCP and have the changes honored, or if it would be expected that changes would go through ACM. I think it's good to be flexible and take your implementation at least for now and we can revisit when we add in the Compute CR etc that Tim is designing.

Reply:
Update pushed and ready for re-review @skeeey. Thanks!